canonical-kubernetes/steps/02_get-kubectl/after-deploy

```diff
@@ -2,28 +2,37 @@
 
 set -eux
 
-. $CONJURE_UP_SPELLSDIR/sdk/common.sh
+. "$CONJURE_UP_SPELLSDIR/sdk/common.sh"
 
-mkdir -p $HOME/.kube
-mkdir -p $HOME/bin
-
-juju scp -m $JUJU_CONTROLLER:$JUJU_MODEL kubernetes-master/0:config ~/.kube/config.$JUJU_MODEL
+mkdir -p "$HOME/.kube"
 
 if [[ $(uname -s) = "Darwin" ]]; then
     KUBE_DEST="/usr/local/bin"
     KUBE_REPO=https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/darwin/amd64
 
-    curl -sLO $KUBE_REPO/kubectl
+    curl -sLO "$KUBE_REPO/kubectl"
     chmod +x kubectl
-    mv kubectl $KUBE_DEST/kubectl
+    mv kubectl "$KUBE_DEST/kubectl"
 else
     KUBE_DEST="/snap/bin"
 
     sudo snap install kubectl --classic || true
 fi
 
-echo "KUBECONFIG=$HOME/.kube/config.$JUJU_MODEL $KUBE_DEST/kubectl \$@" > $HOME/bin/kubectl
-chmod +x $HOME/bin/kubectl
-
-setResult "The Kubernetes client utility (kubectl) is now available"
+# copy down config file from cluster
+juju scp -m "$JUJU_CONTROLLER:$JUJU_MODEL" kubernetes-master/0:config "$HOME/.kube/config.$JUJU_MODEL"
+# change cluster and context name to model name
+sed -i -e "s/juju-\(context\|cluster\)/$JUJU_MODEL/g" "$HOME/.kube/config.$JUJU_MODEL"
+# merge with any existing config and dump to config.new
+KUBECONFIG="$HOME/.kube/config:$HOME/.kube/config.$JUJU_MODEL" "$KUBE_DEST/kubectl" config view --raw=true > "$HOME/.kube/config.new"
```

Comment:
Interesting. I just gave this a try, but one of the admin user credentials was lost in the merge. Can you check this?

I think you're gonna have to change the user name to avoid a collision. So given these two sections:

```
contexts:
- context:
    cluster: juju-cluster
    user: admin
  name: juju-context
users:
- name: admin
  user:
    password: foo
    username: admin
```

You would change `user: admin` and `name: admin`, but not `username: admin` since that's the actual credential.